sharedbuffers/mapped_struct.py

```diff
@@ -745,15 +750,15 @@ def pack_into(cls, obj, buf, offs, idmap = None, implicit_offs = 0,
                         # inline unsigned ints
                         dtype = b'I'
                         buf[offs] = ord(dtype)
-                    elif (cython.cast(cython.longlong, long(-0x8000000000000000)) <= iminval
-                            and imaxval <= cython.cast(cython.longlong, long(0x7FFFFFFFFFFFFFFF))):
+                    elif (cython.cast(cython.longlong, -0x8000000000000000) <= iminval
+                            and imaxval <= cython.cast(cython.longlong, 0x7FFFFFFFFFFFFFFF)):
                         # inline signed int64 list
                         buf[offs] = ord('q')
                         dtype = b'l'
                     else:
                         raise OverflowError
                 except OverflowError:
-                    if 0 <= minval and maxval <= long(0xFFFFFFFFFFFFFFFF):
+                    if 0 <= minval and maxval <= cython.cast(cython.longlong, 0xFFFFFFFFFFFFFFFF):
```

Comment:
longlong can't handle that literal. You need an ulonglong, and IIRC using an ulonglong generates ugly code, so a python long is a better choice. I think using simply `0xFFFFFFFFFFFFFFFF` should work in both versions, but do verify with the annotated code.

Reply:
ping